### PR TITLE
Add osx rid to coreclr projs

### PR DESCRIPTION
**Pull request by agocke**
This sets us up to do coreclr bootstrapping on OS X as well.

@dotnet/roslyn-compiler @jasonmalinowski @davkean Please review

/cc @davidfowl

### CODE REVIEW

File: src/EditorFeatures/CSharpTest2/project.lock.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
           "lib/net45/BasicUndo.dll": {}
         }
       },
-      "Microsoft.CodeAnalysis.Test.Resources.Proprietary/1.1.0-beta1-20150824-02": {
+      "Microsoft.CodeAnalysis.Test.Resources.Proprietary/1.2.0-beta-20151016-11": {
```
Comment (agocke) [inline src/EditorFeatures/CSharpTest2/project.lock.json]: I committed this file by mistake, but this does need to be changed.

**Comment by jasonmalinowski** [comment]
:+1:. I assume that the osx.10.10 RID is something the packages already know about and you've confirmed that's what to use by somebody.

**Comment by agocke** [comment]
@jasonmalinowski Correct.

**Comment by Pilchie** [comment]
:+1:

**Pull request merged**